MetaMask/metamask-extension
# Created the NFT component for setApprovalForAll

## Explanation

Created the `NFT component` for `setApprovalForAll` screen.  Also added `NFT component` to `setApprovalForAll` confirmation screen (when `giving` a setApprovalForAll allowance and when `revoking` a setApprovalForAll allowance).

## More Information

* Fixes #15752
* Fixes #15754 

## Screenshots/Screencaps

### Before

#### Test-dapp (when giving a setApprovalForAll allowance and when revoking a setApprovalForAll allowance)

https://user-images.githubusercontent.com/92527393/225316562-b936a01a-1893-4da0-a3c5-7c1935b4a527.mov

#### Etherscan ( Name of collection and icon `are` implemented in NFTs smart contracts )

https://user-images.githubusercontent.com/92527393/225316953-01b8afcd-6f9a-4c73-b655-cc9abff3aed9.mov

#### Etherscan ( Name of collection and icon `aren't` implemented in NFTs smart contracts )

https://user-images.githubusercontent.com/92527393/225317175-2b1a7afa-37be-4469-963c-2f7284330a92.mov

### After

#### Test-dapp (when `giving` a setApprovalForAll allowance and when `revoking` a setApprovalForAll allowance)

https://user-images.githubusercontent.com/92527393/225313094-569d6d4e-4719-4055-a121-a6219599eb48.mov

#### Etherscan ( Name of collection and icon `are` implemented in NFTs smart contracts )

https://user-images.githubusercontent.com/92527393/225313380-1dac421b-ecd3-4a1b-a897-590f1bb3728c.mov

#### Etherscan ( Name of collection and icon `aren't` implemented in NFTs smart contracts )

https://user-images.githubusercontent.com/92527393/225313690-bd34a959-e46b-492b-a088-646e439c867d.mov

#### Name of collection and icon `are` implemented in NFTs smart contracts

![Screenshot 2023-03-13 at 17 26 34](https://user-images.githubusercontent.com/92527393/224764785-ca924a0f-1c0d-4640-9ca5-5d375d41aafc.png)

#### Name of collection and icon `aren't` implemented in NFTs smart contracts

![Screenshot 2023-03-13 at 17 30 21](https://user-images.githubusercontent.com/92527393/224765644-3def257a-84e1-489b-9fb0-53f90706913e.png)

## Steps to Reproduce

### Note (only for test-dapp): 
If user, when interact with `Mint` in  https://metamask.github.io/test-dapp/, insert value grater that `1`, that value will display in this component (for example, if user insert `4` in the `Mint` input field it will display `All (4) + Future NFTs` in the component.

### Test-dapp

1. Open the test dapp https://metamask.github.io/test-dapp/
2. Under the collectibles section, click on `Deploy`
3. Wait for transaction to be confirmed and click on `Mint` (user can insert grater than `1` in the input field)
4. Wait for transaction to be confirmed and click on `Set Approval For All` -- user can see this component
or
5.  Wait for transaction to be confirmed and click on `Revoke` -- user can see this component

### Etherscan ( Name of collection and icon `are` implemented in NFTs smart contracts )

1. Go to https://etherscan.io/address/0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d#writeContract (erc721)
2. Connect your wallet
3. Go to `setApprovalForAll` and input a random address and write `true` in `approved (bool)` input for `giving` a setApprovalForAll allowance
or
4. Go to `setApprovalForAll` and input a random address and write `false` in `approved (bool)` input for `revoking` a setApprovalForAll allowance
6. Click on `Write`
7. See how it is displays this component

### Etherscan ( Name of collection and icon `aren't` implemented in NFTs smart contracts )

1. Go to https://etherscan.io/token/0x54f456b544abfb785694400bcb1d85629b2d437f#writeContract (erc1155)
2. Connect your wallet
3. Go to `setApprovalForAll` and input a random address and write `true` in `approved (bool)` input for `giving` a setApprovalForAll allowance
or
4. Go to `setApprovalForAll` and input a random address and write `false` in `approved (bool)` input for `revoking` a setApprovalForAll allowance
5. Click on `Write`
6. See how it is displays this component
